### PR TITLE
feat(DHT): validate explicit peer descriptor nodeId length

### PR DESCRIPTION
**Pull request by juslesan**
## Summary

Validate length of the nodeId if an explicit PeerDescriptor is given to the DhtNode.

### CODE REVIEW

File: packages/trackerless-network/test/utils/utils.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export const createStreamMessage = (
 }
 
 export const createRandomNodeId = (): DhtAddress => {
-    return getDhtAddressFromRaw(randomBytes(10))
+    return getDhtAddressFromRaw(randomBytes(20))
```
Comment (harbu) [inline packages/trackerless-network/test/utils/utils.ts]: Is this same as Kademlia ID, so should we do?
```suggestion
    return getDhtAddressFromRaw(randomBytes(KADEMLIA_ID_LENGTH_IN_BYTES))
```

File: packages/dht/src/dht/DhtNode.ts [hunk 2]
```diff
@@ -157,6 +157,11 @@ export class DhtNode extends EventEmitter<Events> implements ITransport {
                 throw new Error(`Invalid nodeId, the length of the nodeId should be ${expectedNodeIdLength}`)
             }
         }
+        if (this.config.peerDescriptor !== undefined) {
+            if (this.config.peerDescriptor.nodeId.length !== KADEMLIA_ID_LENGTH_IN_BYTES) {
```
Comment (harbu) [inline packages/dht/src/dht/DhtNode.ts:161]: peerDescriptor.nodeId should be `KADEMLIA_ID_LENGTH_IN_BYTES` but above in the code this.config.nodeId should be `KADEMLIA_ID_LENGTH_IN_BYTES * 2`. Not sure I follow, should these be the same?

Comment (juslesan) [reply packages/dht/src/dht/DhtNode.ts:161]: The type is different. If a nodeId is given explicitly is given in string as hex. In the PeerDescriptor it is a Uint8Array.

Comment (harbu) [reply packages/dht/src/dht/DhtNode.ts:161]: Got ya

**Review by teogeb — APPROVED** [review]

**Review by harbu — APPROVED** [review]

**Pull request merged**